### PR TITLE
Spelling of develdocs.phtml reworked

### PR DESCRIPTION
**Pull request by comradekingu**
Everything is full of stuff like this, so dipping my toe in to see how hard it could be to do something about it.

### CODE REVIEW

**Comment by fellen** [comment]
Hello, where is the develdoc configuration? ;-)
Perhaps you can reworden the title?

You know that this commit will break the translation?
So I am inclined to delay this PR as we just started to use weblate tor the translation. If we can  attract translators, we can apply it.

Don't be shy to remember me in case I should forget it.

**Comment by comradekingu** [comment]
@fellen Develdocs is what the HTML-file is called, just opened that one because I knew about one error, and then I saw many others. Do you want me to make one PR per file, or keep piling on stuff here?
Weblate handles fuzzy strings correctly, so no translations are lost.
You attracted me so far. Translators like it when things improve.
I'll keep your memory alive.

File: develdocs.phtml [hunk 1]
```diff
@@ -68,11 +68,11 @@ include("externals/menu.phtml");
 </div>
 <p class="justify"><?php echo T_("Get a copy of the documentation source as described above and start making changes. When you are satisfied with your changes you can create a patch by running the following command in the base directory of the documentation (usually gnucash-docs, unless you renamed it):");?></p>
 <pre class="command">git diff > my-patch-name.patch</pre>
-<p class="justify"><?php echo T_("Next attach your patch to a bug report against the proper component for the Documentation product in <a href='https://bugs.gnucash.org/'><span class=\"gnucash\">GnuCash</span>' bug database</a>. See also <a href='https://wiki.gnucash.org/wiki/Bugzilla'>GnuCash' wiki page on bugzilla</a> for more details about patch submission.");?></p>
+<p class="justify"><?php echo T_("Next, attach your patch to a bug-report against the proper component for the \"Documentation\" product in the <a href='https://bugs.gnucash.org/'><span class=\"gnucash\">GnuCash</span>' bug database</a>. See also <a href='https://wiki.gnucash.org/wiki/Bugzilla'>GnuCash' wiki page on bugzilla</a> for more details about patch submission.");?></p>
```
Comment (fellen) [inline develdocs.phtml:71]: Better: `<q>Documentation</q>`, see https://wiki.gnucash.org/wiki/GUI_Guidelines#Quoting_in_Strings.
Note: Most parts of the website are ways older than our GUI Guidelines, but for changes they are valid.

**Comment by fellen** [comment]
@comradekingu I added a bunch of comments to nb.po in https://github.com/Gnucash/gnucash/pull/838/files.

@jralls We need really to configure it different.

File: develdocs.phtml (hunk 1 above)
Comment (comradekingu) [inline develdocs.phtml:71]:
```suggestion
<p class="justify"><?php echo T_("Next, attach your patch to a bug-report against the proper component for the <b>Documentation</b> product in the <a href='https://bugs.gnucash.org/'><span class=\"gnucash\">GnuCash</span>' bug database</a>. See also <a href='https://wiki.gnucash.org/wiki/Bugzilla'>GnuCash' wiki page on Bugzilla</a> for more details about patch submission.");?></p>
```

**Comment by jralls** [comment]
@comradekingu You have an unusual understanding of the word spelling. There isn't a single spelling change in this PR. There are punctuation changes, capitalization changes, and changes to wording, some of which seem no less idiosyncratic than the ones they replace.

> @jralls We need really to configure it different.
@fellen, configure what differently?

File: develdocs.phtml (hunk 1 above)
Comment (fellen) [reply develdocs.phtml]: - thanks!
- bug-report: The English use the hyphen very sparse in contrast to us Continentals.

**Comment by fellen** [comment]
> @fellen, configure what differently?

I think, I understand it slowly, but I would include
- the pot files in the repository
- make pot; msgmerge in the standard ninja/make process

**Comment by comradekingu** [comment]
@jralls Other than those not being the corrected errors, I don't think I am wrong (nor do I really care). Would write "grammar" if it wasn't so brusque.
My aim is to make it easier to translate and read. I translate, see an error, and then go on to fix the most glaring errors of the file it is in.

Yes there are more ways of doing what ultimately is the most proper and final edit right away, but it is a lot of errors, in what amounts to many stages of fixes to get to down the road.

**Comment by fellen** [comment]
I aggree with comradekingu here. Most parts of www were written by an Italian. Getttext offers only a handful of checks with target programming. In contrast the translation toolkit has many checks comparing the structure, linebreaks …
Before we fixed grammatical errors often only in the translation and left the source,
Now the toolkit says: Here seems something be wrong.
Currently I spend much time in telling the translators: do not port the errors! If I know, the are programmers, too, I ask them to send PRs for the source.

So the first result will be an improvement of the source text.
Later it will become easier to translate, more translations …

**Comment by sunfish62** [comment]
Having been goaded to it, I will weigh in here. I agree with John about the changes. I appreciate the changes in capitalization, but I think the textual changes are spurious at best, ungrammatical in one case, and change the tenor, tone, and intent in others.

File: develdocs.phtml [hunk 2]
```diff
@@ -14,7 +14,7 @@ include("externals/menu.phtml");
 
 <p class="justify"><?=$description?></p>
 
-<p class="justify"><?php echo T_("Just as the <span class=\"gnucash\">GnuCash</span> application has been developed over time by volunteers, so too has the documentation been written by volunteer developers and users to help others learn how <span class=\"gnucash\">GnuCash</span> works. This collaborative effort has resulted in the creation of two major documents, the <b>Help Manual</b> and the <b>Tutorial and Concepts Guide</b>.");?></p>
+<p class="justify"><?php echo T_("Just as <span class=\"gnucash\">GnuCash</span> has taken form over time, so too has the documentation to help others learn how it works. There are two major documents, the <b>Help Manual</b> and the <b>Tutorial and Concepts Guide</b>.");?></p>
```
Comment (sunfish62) [inline develdocs.phtml:17]: This changes the original text to a new meaning. The original aims to bring out the volunteer nature of the project; this change eliminate that entirely--a change in intent with which I do not agree.

File: develdocs.phtml [hunk 3]
```diff
@@ -26,8 +26,8 @@ include("externals/menu.phtml");
 <div class="featureborder">
 <h2 class="featuretitle"><?php echo T_("Introduction");?></h2>
 </div>
-<p class="justify"><?php echo T_("The <span class=\"gnucash\">GnuCash</span> documentation is stored in <b>xml</b> files. More precisely, it uses the <b>docbook system</b>. This is a relatively flexible system that takes xml files as input and can generate documentation in several different output formats (html, pdf …).");?></p>
-<p class="justify"><?php echo T_("If you wish to review or write <span class=\"gnucash\">GnuCash</span> documentation, you will need to know xml. Some basic knowledge of the docbook system may be useful as well.");?></p>
+<p class="justify"><?php echo T_("The <span class=\"gnucash\">GnuCash</span> documentation is stored in <b>xml</b> files. More precisely, it uses the <b>docbook system</b>. This is a relatively flexible system that takes xml files as input and can generate documentation in several different output formats (HTML, PDF …).");?></p>
```
Comment (sunfish62) [inline develdocs.phtml:29]: If you're going to capitalize XML, you should do so consistently.

File: develdocs.phtml [hunk 4]
```diff
@@ -68,11 +68,11 @@ include("externals/menu.phtml");
...
+<p class="justify"><?php echo T_("Next, attach your patch to a bug-report against the proper component for the <b>Documentation</b> product in the <a href='https://bugs.gnucash.org/'><span class=\"gnucash\">GnuCash</span>' bug database</a>. See also <a href='https://wiki.gnucash.org/wiki/Bugzilla'>GnuCash' wiki page on Bugzilla</a> for more details about patch submission.");?></p>
 <div class="dialog note">
-<p class="justify"><?php echo T_("<b>NOTE:</b> It used to be ok also to send your patch directly to the gnucash-devel list. This is discouraged now, as a patch is easily forgotten between the many list discussions. Attach patches to bugs in Bugzilla instead (either an existing bug or a new one). If you insist on sending a patch to gnucash-devel, it should be attached, not inlined.");?></p>
+<p class="justify"><?php echo T_("<b>Note:</b> It used to be OK to also send your patch directly to the gnucash-devel list. This is discouraged now, as a patch is easily forgotten between the many list discussions. Attach patches to bugs in Bugzilla instead (either an existing bug or a new one). If you insist on sending a patch to gnucash-devel, it should be attached, not inlined.");?></p>
```
Comment (sunfish62) [inline develdocs.phtml:73]: Your change from "also to send" to "to also send" splits the infinitive form of the verb, and technically is not grammatical. While this locution is now common in the language, this PR is iridium as being targeted at grammar, so this beats noting.

File: develdocs.phtml [hunk 5]
```diff
@@ -68,11 +68,11 @@ include("externals/menu.phtml");
...
 <p class="justify"><?php echo T_("Please let other writers know which section you wish to tackle. Please forward this to <a href='mailto:gnucash-devel@gnucash.org'>gnucash-devel</a> so that people can say 'hey I'm doing that already' or 'go ahead and do it'.");?></p>
-<p class="justify"><?php echo T_("You may also want to retain a local copy of the old documentation to refer to when writing. This still has a lot of useful information in it which hasn't been transferred to the new docs.");?></p>
+<p class="justify"><?php echo T_("You may also want to retain a local copy of the old documentation to refer to when writing. This still has a lot of useful info in it which hasn't been transferred to the new docs.");?></p>
```
Comment (sunfish62) [inline develdocs.phtml:75]: Why make this change?

**Review by sunfish62 — COMMENTED** [review]
My specific criticisms

File: develdocs.phtml [hunk 6]
```diff
@@ -68,11 +68,11 @@ include("externals/menu.phtml");
 </div>
 <p class="justify"><?php echo T_("Get a copy of the documentation source as described above and start making changes. When you are satisfied with your changes you can create a patch by running the following command in the base directory of the documentation (usually gnucash-docs, unless you renamed it):");?></p>
 <pre class="command">git diff > my-patch-name.patch</pre>
-<p class="justify"><?php echo T_("Next attach your patch to a bug report against the proper component for the Documentation product in <a href='https://bugs.gnucash.org/'><span class=\"gnucash\">GnuCash</span>' bug database</a>. See also <a href='https://wiki.gnucash.org/wiki/Bugzilla'>GnuCash' wiki page on bugzilla</a> for more details about patch submission.");?></p>
+<p class="justify"><?php echo T_("Next, attach your patch to a bug-report against the proper component for the <b>Documentation</b> product in the <a href='https://bugs.gnucash.org/'><span class=\"gnucash\">GnuCash</span>' bug database</a>. See also <a href='https://wiki.gnucash.org/wiki/Bugzilla'>GnuCash' wiki page on Bugzilla</a> for more details about patch submission.");?></p>
```
Comment (sunfish62) [inline develdocs.phtml:71]: I don't believe that "bug report" is hyphenated.

Comment (sunfish62) [reply develdocs.phtml:71]: Also, change:

"the proper component for the <b>Documentation</b> product"

To:

"the <b>Documentation</b> component"

Comment (fellen) [reply develdocs.phtml:71]: See
- [Browse](https://bugs.gnucash.org/describecomponents.cgi)
- [Components for Documentation](https://bugs.gnucash.org/describecomponents.cgi?product=Documentation)

Comment (fellen) [reply develdocs.phtml:71]: Another note. This section is - like several pages in the wiki - outdated. Today we want:

1. If possible: Pull Requests at Github,
2. else: patches at bugzilla.
3. No longer recommend as they are often forgotten: patches via gnucash-devel.

Comment (sunfish62) [reply develdocs.phtml:71]: I see your point in re: terminology. Perhaps refer to Documentation as a section? As in:

"the proper component under the Documentation section in the GnuCash bug database"

Comment (jralls) [reply develdocs.phtml:71]: Bugzilla groups bugs by **product** and **component**. The **products** in GnuCash's BZ are `Bugzilla`, `Documentation`, `GnuCash`, `Packaging`, and `Website`. See the [New Bug Page](https://bugs.gnucash.org/enter_bug.cgi). It might seem odd to call Documentation a product, but calling it something else would be confusing.

Comment (fellen) [reply develdocs.phtml:71]: They are not consistent. In Browse they use **product category**.

Comment (fellen) [reply develdocs.phtml:71]: > "the proper component under the Documentation section in the GnuCash bug database"

3 steps backwards to execute sounds confusing.
How about  converting the sentence in a numbered list:

1. open "https://bugs.gnucash.org/enter_bug.cgi"
2. select product "Documentation"
3. select the proper component …

Comment (jralls) [reply develdocs.phtml:71]: > They are not consistent. In Browse they use product category.

But only there. It's just *product* everywhere else, and nowhere is it section.

Comment (sunfish62) [reply develdocs.phtml:71]: Given the bugzilla nomenclature, the original should be used. Sorry for the confusion.

**Comment by sunfish62** [comment]
In further consideration, I would suggest that this page be stripped down significantly, if not removed altogether. The information here fits much better in the wiki, and could connect better with the richer resources that are already there. The website really should just encourage users to join the docs team, and then point them to the wiki. IIRC Geert disagreed with my analysis on this point a number of years back, but I haven't seen a huge influx of writers in the intervening years...

**Comment by jralls** [comment]
>  I would suggest that this page be stripped down significantly, if not removed altogether. 

+1. 

How to Help, Language Translations, and Git Access point to wiki pages. This one should too.

Sizing should be removed.

**Comment by fellen** [comment]
> > I would suggest that this page be stripped down significantly, if not removed altogether.
> 
> +1.

I see it different. Having seen Atlanta leaving the net several times this year and assuming it will appen more often because of the climate change, at least basic information should be stored at more than one place.

**Comment by jralls** [comment]
The problem with that is that the wiki gets frequent updates and the website, aside from releases, gets update every 10 years or so.

As I pointed out, 3 of the 4 recruiting links already point to the wiki. What's special about Documentation?

**Comment by fellen** [comment]
> the wiki gets frequent updates

Only the pages, which I have in view. Like ht-docs many wiki pages still say attach patches to bugzilla or report problems to gnucash-devel

Another benefit if www: it is multilingual, while our wiki is at best bilingual.

**Comment by sunfish62** [comment]
Frank: a few points. First, using the example of the wiki still instructing people to attach patches to bug reports is spurious. I haven't heard anyone on the team say that patches are no longer valid, or that they cannot be attached to bug reports; I've only seen devs say they prefer to have pull requests to those other methods. So the wiki isn't wrong-- it just doesn't reflect current preferences. (And given that it's a wiki, why haven't you followed your own advice and made the changes there?)

Similarly, the argument that the website is multilingual while the wiki is barely bilingual misses on two points. First, this page is intended to encourage users to write documentation. But my sense is that there are really only English and German contributors to the docs, and that most actual writing is taking place in English, with those being translated actively into German. Therefore, this page-- regardless of its multilingual status-- has not succeeded in bringing in writers who speak or write in other languages. Second, the documentation itself is basically bilingual. From my scanning, the only active versions are English and German, with moribund versions in Portuguese, Italian, and Japanese. A comparison of the list traffic for the various languages on the list server shows a similar bilingual bias: the English list has the most traffic by far (showing hundreds of kbs per month), with the German list registering a tenth of that, and the others having barely a trickle. So relegating this to the wiki doesn't really limit anything. I'll also note that documentation in any other language is likely to begin as a direct translation from the English, which will require that author to be fluent in English. 

Next, I believe that the website's primary function is to inform the world what gnucash is and how to get it. Its primary target is the person seeking a better alternative to quicken or quickbooks. The site content supports my belief, with a tone that is friendly, brief, and superficially informative. Anyone who decides that the documentation needs improvement, however, will be an established user. They will have already installed the software, ranted on the user lists (in English) about how difficult gnucash is, been told to RTFM, done that, and then complained about how misleading and dense the docs are. Then, and only then, are they remotely likely to dare enter the special hell that is writing gnucash documentation-- at which point, they will be told to read the wiki on writing documentation. 

To sum up: this page is a failure on its own merits, and would be better placed on the wiki where comradekingu could make changes quickly and fellen could revert them just as easily.

**Comment by fellen** [comment]
> Same procedure as last year?
> Same procedure as every year! 

from [Dinner_for_One](https://en.wikipedia.org/wiki/Dinner_for_One)
10 lines on grammar and 100 lines about removing. ;-)

**Comment by sunfish62** [comment]
Ah, I've been Franked again! I should know better. 

I should have known that you'd raise issues and then obstinately refuse to listen to the ensuing discussion. I should have known that once someone raised points for real consideration, that you'd back off with a demeaning little gotcha comment at the end, as if the whole thing was just a game for you. 

What really galls me is that I fell for it again. I should have known better, but I was duped by the fact that you Franking asked me to weigh in on this. Frank you! 

For the record: if the internet fails because global warming has caused some societal cataclysm, I can pretty much Franking guarantee that writing the documentation for this project is not going to be high on anyone's priorities.

File: develdocs.phtml (hunk 5 above)
Comment (comradekingu) [reply develdocs.phtml:75]: Shorter, meaning less "block-of-text", in turn inducing stress for readers. Smaller gain in the bigger scheme of things. "Info" is informal.

File: develdocs.phtml (hunk 3 above)
Comment (comradekingu) [inline develdocs.phtml:29]:
```suggestion
<p class="justify"><?php echo T_("The <span class=\"gnucash\">GnuCash</span> documentation is stored in <b>XML</b> files in the relatively flexible <b>DocBook system</b>. XML files are used as input, generating documentation in several possible output formats (HTML, PDF …).");?></p>
```

File: develdocs.phtml (hunk 4 above)
Comment (comradekingu) [reply develdocs.phtml:73]: Was mostly trying to avoid the lowercase "ok". Was a bit strange before, so will change properly.

Comment (comradekingu) [inline develdocs.phtml:73]:
```suggestion
<p class="justify"><?php echo T_("<b>Note:</b> Previously, patches could be sent directly to the gnucash-devel list. This is now discouraged, as those are easily forgotten between the many list discussions. Attach patches to bug reports in Bugzilla instead (either in an existing or new bug report). If you insist on sending your patch to gnucash-devel, attach it instead of inlining it.");?></p>
```

File: develdocs.phtml (hunk 2 above)
Comment (comradekingu) [reply develdocs.phtml:17]: The whole sentence (now) 
> Just as <span class=\"gnucash\">GnuCash</span> has taken form over time, so too has the documentation to help others learn how it works.

Doesn't do anything, if "volunteer" is moved to the next sentence, describing the two components.

Comment (comradekingu) [inline develdocs.phtml:17]:
```suggestion
<p class="justify"><?php echo T_("There are two parts to the documentation; <b>Help</b> and <b>Concepts</b>, both of which are written by volunteers.");?></p>
```

File: develdocs.phtml (hunk 6 above)
Comment (comradekingu) [inline develdocs.phtml:71]:
```suggestion
<p class="justify"><?php echo T_("Next, attach your patch to a bug report against the proper component for the <b>Documentation</b> product in the <a href='https://bugs.gnucash.org/'><span class=\"gnucash\">GnuCash</span>' bug database</a>. See also <a href='https://wiki.gnucash.org/wiki/Bugzilla'>GnuCash' wiki page on Bugzilla</a> for more details about patch submission.");?></p>
```

**Comment by gjanssens** [comment]
> In further consideration, I would suggest that this page be stripped down significantly, if not removed altogether. The information here fits much better in the wiki, and could connect better with the richer resources that are already there. The website really should just encourage users to join the docs team, and then point them to the wiki. IIRC Geert disagreed with my analysis on this point a number of years back, but I haven't seen a huge influx of writers in the intervening years...

@sunfish62 Just for the record, I may have disagreed years ago. And that was in the context of an idea I had to port the whole website into a CMS (think WordPress, or my favourite back then - drupal). I have let go of that idea for various reasons. I'm now inclined to see the website as a first presentation of the project, which should be light and concise. Anyone who's interested in more, can be redirected to more elaborate pages on the wiki.

As for the translation argument - the website is translated where the wiki is only two languages at most. I have a plenty of thoughts on that as well, but let me try to keep it short.

1. I think the website lends itself to translation because
  * it's what potential future users are most likely to run into first. It helps if they get this first encounter in their own language.
  * its content is fairly static and it helps translators to keep it that way. That's a good argument to move changing information elsewhere.
2. For the same reasons I think translatable documentation that's fairly static is a good thing.
3. Aside from users we also need contributors. As these can come from all over the world and need to be able to communicate, the default language in the community is English (yes, there are subcommunities in other languages). So anything related to helping people contribute is best in English to reach an audience as wide as possible.
4. In an ideal world we have plenty of contributors with plenty of time on their hands. In that case there would be no reason not to translate everything in all languages of the world, including the wiki pages. We live however in a reality of very little available volunteer time, so it's best spent wisely. So there I think that except for things that are highly localized, we should stick to only one language for anything related to contributing. And as English is currently the de facto lingua franca of the world that is the obvious choice.

**Comment by fellen** [comment]
> In further consideration

I am not against them, but we have better places like

- the mailing lists to have fundamental discussions and

- the wiki to draw concepts.

Pull requests are no appropriate place for them, I fear you are scaring off contributors and other reviewers with them.

**Comment by gjanssens** [comment]
> Next, I believe that the website's primary function is to inform the world what gnucash is and how to get it. Its primary target is the person seeking a better alternative to quicken or quickbooks.

_Sidenote_
I mostly agree. Though as an open source project the website ideally should also be our billboard to advertise for new contributors. It should not only say what gnucash is and how to get it, it should also make clear we invite people to contribute. Not necessarily with the technical details of how that is done. It should be enough each visitor is made aware they *can* contribute in some way and we should provide easy to follow links to further details in the wiki for those that are interested.

Note this is mostly a nitpick and not too relevant to this PR. The website already does this fairly well IMO. I'm sure a professional marketeer could probably rework this to be significantly more efficient. Unfortunately that kind of people rarely does volunteer work so what we have will do.
_End sidenote_

I follow your and @jralls ' assessment this page is inconsistent with the rest of the developer pages. The others all refer to the wiki. I don't see why this one wouldn't.

**Comment by gjanssens** [comment]
Another PR that has been left lingering for more than 2 years. Is this one worth discussing further or should we just act on the suggestion to replace this page with a link to the wiki ?

**Comment by jralls** [comment]
IMO there's no point in beating the horse any more. Let's move the info to the wiki and point to that from the website.

**Comment by gjanssens** [comment]
Agreed. Which also means there's no point in keeping this PR open.

**Comment by gjanssens** [comment]
For reference, all relevant info from develdocs.phtml was already included in wiki/Documentation_Improvements so I have linked to that page instead.
